templates/default/github_pr_comment_trigger.rb.erb

```diff
@@ -0,0 +1,166 @@
+end
+
+# Make sure the commenter has permission to perform the merge. The user has
+# permission if their username is in AUTHORIZED_USERS, if one of the
+# organizations they are in is in AUTHORIZED_ORGS, or if one of the teams they
+# are in is in AUTHORIZED_TEAMS. If AUTHORIZED_USERS, AUTHORIZED_ORGS, and
+# AUTHORIZED_TEAMS are all empty, then everyone has permission.
+user = d['comment']['user']['login']
+unless (AUTHORIZED_USERS.empty? &&
+       AUTHORIZED_ORGS.empty? &&
+       AUTHORIZED_TEAMS.empty?) ||
+       AUTHORIZED_USERS.include?(user) ||
+       AUTHORIZED_ORGS.detect { |o| github.organization_member?(o, user) } ||
+       AUTHORIZED_TEAMS.detect { |t| team_member?(github, t, user) }
+  abort "Error: Cannot merge PR because user '#{user}' is not authorized."
+end
+
+# Merge the PR
+github.merge_pull_request(repo_path, issue_number)
+
+# Pull the updated target branch down
+Git.configure do |config|
+  config.binary_path = '<%= @git_path %>'
+end
+git = Git.open('.')
+git_branch = pr['base']['ref']
+git.branch(git_branch).checkout
+git.pull(git.remote('origin'), git_branch)
+
+# Bump the cookbook version in metadata.rb
+# Match the line that looks like `version   "1.2.3"`
+version = '' # Get the version variable in scope
+version_regex = /^(version\s+)(["'])(\d+\.\d+\.\d+)\2$/
+md = ::File.read(METADATA_FILE).gsub(version_regex) do
+  key = Regexp.last_match(1) # The "version" key and some whitespace
+  quote = Regexp.last_match(2) # The type of quotation mark used, e.g. " vs '
+  version = inc_version(Regexp.last_match(3), LEVELS[level])
+  # Reconstruct the version line by using the new version with the same spacing
+  # and quotation mark types as before
+  "#{key}#{quote}#{version}#{quote}"
+end
+::File.write(METADATA_FILE, md)
+
+# Update the CHANGELOG.md with the PR's title
+entry = "#{version} (#{Time.now.strftime('%Y-%m-%d')})"
+entry += "\n" + '-' * entry.length
+entry += "\n- #{d['issue']['title']}\n\n"
+# Inject the new entry above the first one we find
+cl = ::File.read(CHANGELOG_FILE).sub(/^(.*\d+\.\d+\.\d+)/, entry + '\1')
+::File.write(CHANGELOG_FILE, cl)
+
+# Commit changes
+git.add(all: true)
+git.commit("Automatic #{level}-level version bump to v#{version} by Jenkins")
+
+# Create a version tag
+git.add_tag("v#{version}")
+
+# Push back to Github
+git.push(git.remote('origin'), git_branch, tags: true)
+
+# Upload to the Chef server
+$stderr.puts "Uploading #{repo_name} cookbook to the Chef server..."
+# `knife cookbook upload -o ../ #{repo_name} --freeze`
+
+# Close the PR
+message = "Jenkins has merged this PR into `#{git_branch}` and has " \
+  "automatically performed a #{level}-level version bump to v#{version}.  " \
+  'Have a nice day!'
+github.add_comment(repo_path, issue_number, message)
+
+# Return some environment variables for the Chef environment bumper job to use.
+# If we got no envs, the file won't be created and the job won't be triggered.
+unless envs.nil?
+  File.write('envvars',
```

Comment:
The previous `File.write` command was namespaced with `::File`. Is there a reason this one is not?